src/libraries/System.Security.Cryptography.Cose/src/System/Security/Cryptography/Cose/CoseMessage.cs

```diff
@@ -46,37 +48,34 @@ internal CoseMessage(CoseHeaderMap protectedHeader, CoseHeaderMap unprotectedHea
             }
         }
 
-        public static CoseSign1Message DecodeSign1(byte[] cborPayload)
+        public static CoseSign1Message DecodeSign1(ReadOnlySpan<byte> cborPayload)
         {
             try
             {
-                var reader = new CborReader(cborPayload);
-                CborTag? tag = DecodeTag(reader);
-                if (tag != null && tag != Sign1Tag)
+                unsafe
                 {
-                    throw new CryptographicException(SR.Format(SR.DecodeSign1IncorrectTag, tag));
-                }
+                    fixed (byte* ptr = &MemoryMarshal.GetReference(cborPayload))
+                    {
+                        using MemoryManager<byte> manager = new PointerMemoryManager<byte>(ptr, cborPayload.Length);
```

Comment:
Let's keep the scope abundantly clear, and not use statement/braceless usings, please.